kedify-agent/templates/agent-rbac.yaml

```diff
@@ -0,0 +1,449 @@
+  - get
+  - delete
+  - update
+  resourceNames:
+  - kedify-agent
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - delete
+  - update
+  resourceNames:
+  - kedify-agent
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kedify-manager-rolebinding
+  namespace: {{ .Release.Namespace }}
+  labels:
+    control-plane: kedify-agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kedify-manager-role
+subjects:
+- kind: ServiceAccount
+  name: kedify-agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kedify-manager-rolebinding
+  namespace: {{ .Release.Namespace }}
+  labels:
+    control-plane: kedify-agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kedify-manager-role
+subjects:
+- kind: ServiceAccount
+  name: kedify-agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kedify-manager-role
+rules:
+
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - metrics.k8s.io
+  resources:
+  - pods
+  verbs:
+  - get
+  - list
+
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+  - list
+
+- apiGroups:
+  - autoscaling
+  resources:
+  - horizontalpodautoscalers
+  verbs:
+  - get
+  - list
+  - watch
+
+- apiGroups:
+  - apps
+  resources:
+  - statefulsets
+  verbs:
+  - get
+  - list
+  - watch
+
+# required by Kedify to be able to crud CR(D)s related to KEDA
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - delete
+  - get
+  - patch
+  - update
+  resourceNames:
+  - kedifyconfigurations.install.kedify.io
+  - scalingpolicies.keda.kedify.io
+  - clustertriggerauthentications.keda.sh
+  - cloudeventsources.eventing.keda.sh
+  - clustercloudeventsources.eventing.keda.sh
+  - httpscaledobjects.http.keda.sh
+  - scaledjobs.keda.sh
+  - scaledobjects.keda.sh
+  - triggerauthentications.keda.sh
+- apiGroups:
+    - apiextensions.k8s.io
+  resources:
+    - customresourcedefinitions
+  verbs:
+    - create
+- apiGroups:
+  - keda.sh
+  - http.keda.sh
+  - eventing.keda.sh
+  - install.kedify.io
+  - keda.kedify.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
+- apiGroups:
+  - apiregistration.k8s.io
+  resources:
+  - apiservices
+  verbs:
+  - delete
+  - get
+  - patch
+  - update
+  resourceNames:
+  - v1alpha1.keda.kedify.io
+  - v1alpha1.keda.sh
+  - v1alpha1.http.keda.sh
+  - v1alpha1.eventing.keda.sh
+  - v1beta1.external.metrics.k8s.io
+  - keda-operator-metrics-apiserver
+- apiGroups:
+  - apiregistration.k8s.io
+  resources:
+  - apiservices
+  verbs:
+  - list
+  - create
+  - watch
+
+# Kedify needs to configure envoy proxy also in other namespaces & deploy it
```

Comment:
I'm affraid we may need to hide this behind a flag "ingress autowiring". I realize we can't `list` or `watch` with `resourceName`. Maybe we could at least move the `create` to the narrower `deployment` rule just for `kedify-proxy`

Reply:
resourceNames + verb=create won't compile (https://stackoverflow.com/a/65203104)